commands/apps_dev.go

```diff
@@ -397,3 +413,66 @@ func appDevWorkspace(cmdConfig *CmdConfig) (*workspace.AppDev, error) {
 		AppsService:       cmdConfig.Apps(),
 	})
 }
+
+// PrepareEnvironment pulls required images, validates permissions, etc. in preparation for a component build.
+func appDevPrepareEnvironment(ctx context.Context, ws *workspace.AppDev, cli builder.DockerEngineClient, componentSpec godo.AppBuildableComponentSpec) error {
+	cnbBuilderImage := builder.CNBBuilderImage
```

Comment:
We need to pre-pull the `apps-run` image; not sure the best way to infer this info based on the `CNBBuilderImage` ; perhaps hard code temporarily but we should be able to fetch it from the image metadata of the builder after we pull it.